perfact/zodbsync/subcommand.py

```diff
@@ -113,6 +113,64 @@ def check_repo(self):
             if line.endswith(' HEAD')
         ][0].split()[0]
 
+    @staticmethod
+    def gitexec(func):
+        """
+        Decorator for wrapping an operation with playback:
+        - Stash unstaged changes away
+        - memorize the current commit
+        - do something
+        - check for conflicts
+        - play back changed objects (diff between old and new HEAD)
+        - unstash
+        """
+        @SubCommand.with_lock
+        def wrapper(self, *args, **kwargs):
+            # Check for unstaged changes
+            self.check_repo()
+
+            try:
+                func(self, *args, **kwargs)
+                # Are these all?
+                for fname in ['rebase-merge', 'rebase-apply',
+                              'CHERRY_PICK_HEAD', 'MERGE_HEAD', 'REVERT_HEAD']:
```

Comment:
These rascals should live in a named list, making this check more understandable. Something like `git_state_indicators`, i'm not sure. This list should live in a class or even module variable to allow monkey patching.

Reply:
noted, good idea